api-client/src/main/java/com/xing/api/internal/json/SafeCalendarJsonAdapter.java

```diff
@@ -64,44 +71,25 @@
-    private static final Map<String, DateFormat> DATE_FORMAT_MAP = new LinkedHashMap<>(5);
+    private static final Map<Pattern, DateFormat> DATE_FORMAT_MAP = new LinkedHashMap<>(5);
 
     static {
         DATE_FORMAT_MAP.put(REG_EX_YEAR, new SimpleDateFormat(YEAR_DATE_FORMAT, Locale.ENGLISH));
-        DATE_FORMAT_MAP.put(REG_EX_YEAR_MONTH, new SimpleDateFormat(YEAR_MONTH_DATE_FORMAT, Locale.ENGLISH));
-        DATE_FORMAT_MAP.put(REG_EX_YEAR_MONTH_DAY, new SimpleDateFormat(YEAR_MONTH_DAY_DATE_FORMAT, Locale.ENGLISH));
-        DATE_FORMAT_MAP.put(REG_EX_ISO_DATE_Z, new SimpleDateFormat(ISO_DATE_FORMAT_Z, Locale.ENGLISH));
-        DATE_FORMAT_MAP.put(REG_EX_ISO_DATE_TIME, new SimpleDateFormat(ISO_DATE_FORMAT, Locale.ENGLISH));
+        DATE_FORMAT_MAP.put(REG_EX_YEAR_MONTH,
+                new SimpleDateFormat(YEAR_MONTH_DATE_FORMAT, Locale.ENGLISH));
+        DATE_FORMAT_MAP.put(REG_EX_YEAR_MONTH_DAY,
+                new SimpleDateFormat(YEAR_MONTH_DAY_DATE_FORMAT, Locale.ENGLISH));
+        DATE_FORMAT_MAP.put(REG_EX_ISO_DATE_Z,
+                new ZuluDateFormat(ISO_DATE_FORMAT_Z, Locale.ENGLISH));
+        DATE_FORMAT_MAP.put(REG_EX_ISO_DATE_TIME,
+                new SimpleDateFormat(ISO_DATE_FORMAT, Locale.ENGLISH));
         DATE_FORMAT_MAP.put(REG_EX_THREE_LETTER_ISO8601_DATE_FORMAT,
-              new SimpleDateFormat(ISO_DATE_FORMAT, Locale.ENGLISH) {
-                  @Override
-                  public StringBuffer format(Date date, StringBuffer toAppendTo, FieldPosition pos) {
-                      StringBuffer dateString = super.format(date, toAppendTo, pos);
-                      return dateString.insert(dateString.length() - 2, ':');
-                  }
-
-                  @Override
-                  public Date parse(String text, ParsePosition pos) {
-                      int index = text.length() - 3;
-                      text = text.substring(0, index) + text.substring(index + 1);
-                      return super.parse(text, pos);
-                  }
-              });
-        DATE_FORMAT_MAP.put(REG_EX_ISO_DATE_WEIRD, new SimpleDateFormat(ISO_DATE_FORMAT_WEIRD, Locale.ENGLISH));
+                new ThreeLetterDateFormat(ISO_DATE_FORMAT, Locale.ENGLISH));
+        DATE_FORMAT_MAP.put(REG_EX_ISO_DATE_WEIRD,
+                new ZuluDateFormat(ISO_DATE_FORMAT_WEIRD, Locale.ENGLISH));
     }
 
-    public static final Factory FACTORY = new Factory() {
-        @Override
-        public JsonAdapter<?> create(Type type, Set<? extends Annotation> annotations, Moshi moshi) {
-            if (!annotations.isEmpty()) return null;
-            Class<?> rawType = Types.getRawType(type);
-            if (rawType == SafeCalendar.class || rawType == GregorianCalendar.class
-                  || rawType == Calendar.class) {
-                return new SafeCalendarJsonAdapter().nullSafe();
-            }
-            return null;
-        }
-    };
+    public static final Factory FACTORY = new SafeCalendarFactory();
```

Comment:
No need for `SafeCalendarFactory`, we want to be consistent across all `JsonAdapter` files. If you can please move this to the top of the file.